### PR TITLE
Add Groups, Trusts, and Permissions to the contract manifest

### LINKED ISSUE #348: Add support for trusts and permissions
A contract manifest contains the attributes `trusts`, and `permissions`. The compiler must offer a way to specify these attributes when implementing a Java smart contract.

### LINKED ISSUE #354: Support adding contract groups to contract manifest
The contract manifest has a `group` attribute which is a list of `Group` objects that represent a trusted circle of contracts which can be interacted with without any extra notifications. I.e. contracts in the group can be called by the contract and can call the contract without triggering a notification. 

The neow3j devpack needs to allow setting of such group objects in the contract manifest

### PR DESCRIPTION
**Pull request by gsmachado**
Closes #348 
Closes #354

### CODE REVIEW

File: compiler/src/test/java/io/neow3j/compiler/GroupManifestTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        assertThat(groups.get(1).getSignature(), is("signature2"));
+    }
+
+    @Test
+    public void withoutGroupsAnnotation() throws IOException {
+        CompilationUnit unit = new Compiler().compileClass(GroupManifestTestContractWithoutAnnotation.class.getName());
+        List<ContractGroup> groups = unit.getManifest().getGroups();
+        assertThat(groups, hasSize(0));
+    }
+
```
Comment (csmuller) [inline compiler/src/test/java/io/neow3j/compiler/GroupManifestTest.java:48]: Please add a test case that uses only one `@Group` annotation.

File: compiler/src/test/java/io/neow3j/compiler/PermissionManifestTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    public void withoutPermissionsAnnotation() throws IOException {
+        CompilationUnit unit = new Compiler()
+                .compileClass(PermissionManifestTestContractWithoutAnnotation.class.getName());
+        List<ContractPermission> permissions = unit.getManifest().getPermissions();
+        assertThat(permissions, hasSize(1));
+        assertThat(permissions.get(0).getContract(), is("*"));
+        assertThat(permissions.get(0).getMethods(), hasSize(1));
+        assertThat(permissions.get(0).getMethods().get(0), is("*"));
+    }
+
```
Comment (csmuller) [inline compiler/src/test/java/io/neow3j/compiler/PermissionManifestTest.java:55]: Please add a test case that uses only one `@Permission` annotation.

File: compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java [hunk 3]
```diff
@@ -106,17 +110,92 @@ private static ContractABI buildABI(NeoModule neoModule) {
...
+                methods.addAll((List<String>) methodsValues);
+            }
+
+            ContractPermission contractPermission = new ContractPermission(contract, methods);
+            permissions.add(contractPermission);
+        }
+        return permissions;
+    }
+
+    private static List<ContractGroup> getContractGroups(AnnotationNode ann) {
```
Comment (csmuller) [inline compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java:178]: We could enforce rules on the annotation values here.
- The public key has to be a 33 bytes long hex string.
- The signature has to be Base64 encoded.

File: compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java [hunk 4]
```diff
@@ -106,17 +110,92 @@ private static ContractABI buildABI(NeoModule neoModule) {
...
+            groups.add(cgManifest);
         }
-        AnnotationNode ann = annotationNode.get();
-        List<String> standards = new ArrayList<>();
-        for (Object standard : (List<?>) ann.values.get(1)) {
-            standards.add((String) standard);
+        return groups;
+    }
+
+    private static List<String> getContractTrusts(AnnotationNode ann) {
```
Comment (csmuller) [inline compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java:191]: We could enforce rules on the annotation values here.
The value has to be either a 33 (public key) or 20 bytes long (script hash) hex string.

File: compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java [hunk 5]
```diff
@@ -106,17 +110,92 @@ private static ContractABI buildABI(NeoModule neoModule) {
...
+                .map(ann -> {
+                    List<String> values = new ArrayList<>();
+                    for (Object value : (List<?>) ann.values.get(1)) {
+                        values.add((String) value);
+                    }
+                    return values;
+                });
+    }
+
+    private static List<ContractPermission> getContractPermissions(AnnotationNode ann) {
```
Comment (csmuller) [inline compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java:153]: We could enforce rules on the annotation values here.
The contract value has to be either a 33 (public key) or 20 bytes long (script hash) hex string.

**Review by csmuller — CHANGES_REQUESTED** [review]

File: compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java (hunk 3 above)
Comment (gsmachado) [reply compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java:178]: Good point. I will check that. 👍

File: compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java (hunk 4 above)
Comment (gsmachado) [reply compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java:191]: Yes, good point. I will do that.

File: compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java (hunk 5 above)
Comment (gsmachado) [reply compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java:153]: Yet another one. 😄 
I will implement it.

File: compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java (hunk 3 above)
Comment (gsmachado) [reply compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java]: implemented.

File: compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java (hunk 4 above)
Comment (gsmachado) [reply compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java]: implemented.

File: compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java (hunk 5 above)
Comment (gsmachado) [reply compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java]: implemented.

**Review by csmuller — APPROVED** [review]

**Pull request merged**